WordPress/Classes/Utility/Router.swift

```diff
@@ -0,0 +1,112 @@
+
+import WordPressComStatsiOS
+
+/// `Router` is intended to be used to easily navigate and display common elements natively
+/// like Posts, Site streams, Comments, etc...
+///
+struct Router {
+
+    enum DisplayError: Error {
+        case missingParameter
+        case unsupportedFeature
+        case unsupportedType
+    }
+
+    private let mainContext: NSManagedObjectContext
+    private let controller: UIViewController
```

Comment:
Should this reference be weak? It does not seem to leak at the moment, but my gut feeling is that this has a lot of potential for becoming a reference cycle

Reply:
It's probably not creating a retain cycle because Router is a struct, but I think it makes sense to declare `controller` as weak anyway, so it's safer for future changes. Good catch!